### ISSUE TITLE
Camelot image co-ordinates to PDF box

### ISSUE THREAD

**Issue by Siddharth1India** [camelot-dev/camelot#377]
I am working on this extremely complex problem where almost everything is variable. My final solution is, converting pdf to image, getting table coordinates from image with table-transformers and using those coordinates to get table from PDF and read PDF table with camelot.<br>
Now issue is, I am getting box coordinates from image but they are not same as PDF coordinates and hence I cannot box-out table. 
Note: I am using `pdf2image` for converting PDF to image. 
Thanks

**Comment by vaibhavkansallumiq** [comment]
did you find solution ??

**Comment by Siddharth1India** [comment]
@vaibhavkansallumiq this may help: https://github.com/atlanhq/camelot/issues/486

**Comment by bosd** [comment]
Yes — what you're doing is effectively **manual table-area selection**: you detect the table region with an external image-based tool (table-transformers) and want to hand that region to Camelot as `table_areas`. Camelot fully supports that; the only missing piece is the coordinate conversion, which is the reason the box doesn't line up.

Your detector's box is in **image pixels** (origin top-left), but `table_areas` is in **PDF coordinate space** (origin bottom-left, points). Two conversions are needed: a per-axis **scale** and a **y-flip**. The important gotcha in your case: you rendered with `pdf2image` at your own DPI, so the scale back to PDF points is `72/dpi` — not Camelot's internal 300 dpi.

```python
import camelot

x0_img, y0_img, x1_img, y1_img = detected_box   # px, origin top-left
dpi = 200                                        # your pdf2image dpi
s = 72.0 / dpi                                   # px -> PDF points

page_h_pts = image_height_px * s                 # rendered image height -> points
x0, x1 = x0_img * s, x1_img * s
y_top    = page_h_pts - y0_img * s               # image top -> larger PDF y
y_bottom = page_h_pts - y1_img * s               # image bottom -> smaller PDF y

# table_areas = 'x1,y1,x2,y2' (top-left, bottom-right) in PDF space
area = f'{x0},{y_top},{x1},{y_bottom}'
tables = camelot.read_pdf('doc.pdf', flavor='lattice', table_areas=[area])
```

I've added this (and the reverse direction) to the docs FAQ — see the new *"I have table coordinates from an image — how do I pass them to table_areas?"* entry: https://camelot-py.readthedocs.io/en/latest/user/faq.html (landing via #781).

Since your table appears ruled, `flavor='lattice'` is right; if the rules are faint, the new `engine='combined'` (just merged) will help. Does the conversion above resolve the misalignment?

**Comment by mvanhorn** [comment]
Implemented in https://github.com/camelot-dev/camelot/pull/823 - adds a public `camelot.image_bbox_to_pdf(bbox, image_size, pdf_size)` that applies the y-axis flip plus page-to-image scale to turn an image-pixel bbox into a PDF-space `table_areas` entry.
